### ISSUE TITLE
Dragging windows snapped to side, makes the windows invisible (Sometimes)

### ISSUE THREAD

**Issue by Alfenstein8** [Leleat/Tiling-Assistant#140]
**Describe the bug**
When dragging a window from the extensions' app, snapped to the side will make the window invisible until I place the mouse over the window. It has only happened with the extensions' app.
Here is a video showing the bug: https://odysee.com/Tiling-Assistant-v.32-bug:c 
I get the same result when disabling all other GNOME extensions, and it doesn't happen when Tiling-Assistant is disabled.

**Steps To Reproduce**
1. Snap a window from the extensions' app to the side of the screen.
2. Drag the window with the header or with SUPER + drag

**System Info:**
 - Distro (incl. version): Fedora release 35 (Thirty Five) x8
 - GNOME (Shell) version: GNOME 41.3
 - Extension version and from where (e. g. EGO, `main` branch...): main version 32
 - XOrg/Wayland: Wayland

**Journalctl logs**

**Comment by Leleat** [comment]
I am also on Fedora 35 with GNOME Shell 41.3. I've tested Wayland now and unfortunately I can't reproduce this.

https://user-images.githubusercontent.com/53054301/150664763-c2412962-b08a-4e71-9043-3f93d625e3e5.mov

After enabling the setting to detach modal dialogs from their parent windows the modal dialogs can't be tiled (see video) under Wayland (which I consider a bug and will fix). No idea why it works for you...  Dumb question: are you sure you are on Wayland? So you are also affected by https://github.com/Leleat/Tiling-Assistant/issues/138?

But tbh I am drawing a blank here...

**Comment by Alfenstein8** [comment]
>Dumb question: are you sure you are on Wayland? So you are also affected by #138?

No wtf 🤦. That should be a dumb question, but it's not. This is entirely on me. I thought I was on Wayland this entire time.

**Comment by Leleat** [comment]
> No wtf facepalm. That should be a dumb question, but it's not. This is entirely on me. I thought I was on Wayland this entire time.

Alright 😄 then, I'll try to reproduce it with X.

PS. Did you notice the bug with v31 as well?

**Comment by Alfenstein8** [comment]
> PS. Did you notice the bug with v31 as well?

Yes, I just tried it with 31. Same result

**Comment by Leleat** [comment]
Hmm, can't reproduce this on Xorg either... Kinda stab in the dark:

Can you please

1. disable **all** extensions
2. reload GNOME Shell (Alt+F2 -> r)
3. Re-eable only Tiling Assistant
4. disable the tiling & untiling animations in the [advanced settings](https://github.com/Leleat/Tiling-Assistant/blob/main/GUIDE.md#Hidden-Settings) of Tiling Assistant

Does the issue still presist?

**Comment by taernsietr** [comment]
Hey @Leleat. Sorry if writing under this issue is incorrect, I'm kinda new at GitHub workflows.

But I've had a similar issue, where the windows _do not_ turn invisible but rather sort of stop updating, quickly switching between frames generated at the moment the windows is snapped. The windows/programs are responding (i.e. writing in gedit when the bug occurs will still write to gedit, just not show anything writting), just not graphically updating.

I did attempt the steps you described above and seem to be unable to reproduce after doing so. Are there any steps I could take to provide better information as to what might be causing this?

I did not individually test Tiling Assistant with other extensions that I'm currently using, and I know it's kinda pointless, but I'll list them anyway (maybe people with the same bug are using some of the same?):

- Blur my Shell
- Dash to Panel
- Freon
- GSConnect (did not disable during testing)
- Net Speed Simplified
- Notification Banner Position
- Sound Input & Output Device Chooser
- Tray Icons: Reloaded
- Unblank Lock Screen
- User themes (plain theming, nothing too crazy)
- Window is Read Notification Remover
- windowNavigator

**System Info:**

**Distro (incl. version):** Arch Linux 5.16.4-arch1-1
**GNOME (Shell) version:** GNOME 41.3-1
**Extension version:** 32, Gnome Extensions website
**XOrg/Wayland:** XOrg

Thanks a lot for your work on this extension, by the way!

**Comment by Leleat** [comment]
> Hey @Leleat. Sorry if writing under this issue is incorrect, I'm kinda new at GitHub workflows.

Don't worry about it, you can write here.

> I did attempt the steps you described above and seem to be unable to reproduce after doing so.

Do you mean the ones where you only tested Tiling Assistant? And the bug went away?

**Comment by Alfenstein8** [comment]
> Hmm, can't reproduce this on Xorg either... Kinda stab in the dark:
> 
> Can you please
> 
>     1. disable **all** extensions
> 
>     2. reload GNOME Shell (Alt+F2 -> r)
> 
>     3. Re-eable only Tiling Assistant
> 
>     4. disable the tiling & untiling animations in the [advanced settings](https://github.com/Leleat/Tiling-Assistant/blob/main/GUIDE.md#Hidden-Settings) of Tiling Assistant
> 
> 
> Does the issue still presist?

It did do something different this time. More like what @taernsietr is describing
Video: https://odysee.com/different-:c

**Comment by taernsietr** [comment]
> Do you mean the ones where you only tested Tiling Assistant? And the bug went away?

Yes, I tested it without any extensions other than Tiling Assistant and GSConnect, and the bug didn't seem to occur. Seeing as it doesn't happen 100% of the time, though, I'm can't be sure it wouldn't have.

> Do the apps only visually freeze for a short period or do they never update again?

They don't update unless interacted with, although they don't return to a usable state, there might be visual changes but the glitching continues, or if maximized, in which case they return to a normal state.

Alfenstein8's video shows the bug precisely as I experience it.

**Comment by cachandlerdev** [comment]
I am having the "apps only update when window focus is switched" bug as well. Has there been any progress on this issue?

The only other extensions I'm using are "Alphabetical App Grid" (alphabetically sorts apps) and "AppIndicator and KStatusNotifierItem Support" (adds system tray icons).

**Comment by Leleat** [comment]
> I am having the "apps only update when window focus is switched" bug as well using Fedora 36 with Xorg. Has there been any progress on this issue?
>
>The only other extensions I'm using are "Alphabetical App Grid" (alphabetically sorts apps) and "AppIndicator and KStatusNotifierItem Support" (adds system tray icons).

I wasn't able to reproduce it. So I *think* the issue isn't directly within my extension. If that's the case it's probably not fixable on my end.

Best way to confirm this is for those, who are affected by this issue, to try another tiling extension and see if they encounter a similiar issue. If you don't have this bug then it's an issue with my extension and I could fix it (by comparing my extension to the other tiling extension).

**Comment by SubOptimal** [comment]
I discover a behaviour which might be related.

**Setup**
— only the Tiling Assistant extension is activated, using Gnome 43.3 on Xorg

**reproducer**
— play a video in VLC media player and open Gedit
— activate the layout '4 Quartes'
— position VLC in the upper left and Gedit in the upper right
— as soon Gedit is positioned the video frame rate drops to a few frames per second, audio is still fine
— un-tile the VLC window and the video plays immediate correctly

Even this is not yet reproducible to one hundred percent it happens that way quite often. Something which happens before seems to have an impact on it. Will try to find the reason which trigger it.

The following errors might be of interest. They appeared in the journal log, when the above happened.

```
Mar 25 22:07:16  gnome-shell[2361]: Window manager warning: Invalid WM_TRANSIENT_FOR window 0x2c00008 specified for 0x2c0050c.
Mar 25 22:07:18  gnome-shell[2361]: Window manager warning: WM_TRANSIENT_FOR window 0x2c000a4 for 0x2c000b5 window override-redirect is an override-redirect window and this is not correct according to the standard, so we'll fallback to the first non-override-redirect window 0x2c00006.
Mar 25 22:07:20  gnome-shell[2361]: Can't update stage views actor <unnamed>[<MetaWindowActorX11>:0x5583ef567990] is on because it needs an allocation.
Mar 25 22:07:20  gnome-shell[2361]: Can't update stage views actor <unnamed>[<MetaSurfaceActorX11>:0x5583ec7ab7e0] is on because it needs an allocation.
Mar 25 22:07:32  gnome-shell[2361]: Error in size change accounting.
```
